dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -361,6 +364,12 @@ public T deleteHostedVersion(@ApiParam(hidden = true) @Auth User user,
         checkEntry(entry);
         checkUserCanUpdate(user, entry);
         checkHosted(entry);
+        // If the version that's about to be deleted is the default version, unset it
+        if (entry.getActualDefaultVersion().getName().equals(version)) {
+            Optional<U> max = entry.getWorkflowVersions().stream().filter(v -> Objects.equals(v.getName(), version))
```

Comment:
Shouldn't the filter be `!Object.equals(...`? You're trying to grab the max date of all versions excluding the version you're about to delete?